tests/Functional/ContentEventRenderTest.php

```diff
@@ -183,42 +167,511 @@ public function testEventRegistrationDateDescription(): void {
+    $field_list_expected_values['items'][6] = [
+      'label' => 'Online time',
+      'body' => "18 March 2020, 01:00 AEDT\n to 18 April 2020, 00:00 AEST",
+    ];
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    $icons_text_expected_values['items'][3] = [
+      'icon' => 'livestreaming',
+      'text' => 'Live streaming available',
+    ];
+    $icons_text_assert->assertPattern($icons_text_expected_values, $details_list_content->getOuterHtml());
+
+    // Assert changing type of "Online type".
+    $node->set('oe_event_online_type', 'livestream')->save();
+    $this->drupalGet($node->toUrl());
+
+    $field_list_expected_values['items'][4] = [
+      'label' => 'Live stream',
+      'body' => 'Livestream',
+    ];
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    // Assert "Event website" field.
+    $node->set('oe_event_website', [
+      'uri' => 'http://www.example.com/event',
+      'title' => 'Event website',
+    ])->save();
+    $this->drupalGet($node->toUrl());
+
+    $field_list_expected_values['items'][7] = [
+      'label' => 'Website',
+      'body' => 'Event website',
+    ];
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    // Assert "Registration capacity" field.
+    $node->set('oe_event_registration_capacity', 'event registration capacity')->save();
+    $this->drupalGet($node->toUrl());
+
+    $field_list_expected_values['items'][8] = [
+      'label' => 'Number of seats',
+      'body' => 'event registration capacity',
+    ];
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    // Assert "Entrance fee" field.
+    $node->set('oe_event_entrance_fee', 'entrance fee')->save();
+    $this->drupalGet($node->toUrl());
+
+    $field_list_expected_values['items'][9] = [
+      'label' => 'Entrance fee',
+      'body' => 'entrance fee',
+    ];
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    // Assert "Registration URL" field.
+    $node->set('oe_event_registration_url', 'http://www.example.com/registation');
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    $registration_content = $this->assertSession()->elementExists('css', '#event-registration-block');
+    $this->assertRegisterButtonEnabled($registration_content);
+
+    // Assert "Registration date" field in the past.
+    $registration_start_date = (clone $static_time)->modify('- 2 months');
+    $registration_end_date = (clone $static_time)->modify('- 1 months');
+    $node->set('oe_event_registration_dates', [
+      'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
```

Comment:
See above, when testing time-related features we must never change the actual event dates, we should change time, run cron and reload the page, instead.